examples/web-server/nginx/README.md

```diff
@@ -0,0 +1,148 @@
+# How to use NGINX as a reverse-proxy with LetsEncrypt
+
+## Requirements
+
+1. Start a Coder deployment with a wildcard subdomain. See [this guide](https://coder.com/docs/v2/latest/admin/configure#wildcard-access-url) for more information.
+
+2. Configure your DNS provider to point your YOUR_SUBDOMAIN and \*.YOUR_SUBDOMAIN to your server's public ip.
+
+   > For example, to use `coder.example.com` as your subdomain, configure `coder.example.com` and `*.coder.example.com` to point to your server's public ip. This can be done by adding A records in your DNS provider's dashboard.
+
+3. Install NGINX (assuming you're on Debian/Ubuntu):
+
+   ```console
+   sudo apt install nginx
+   ```
+
+4. Stop NGINX service:
+
+   ```console
+   sudo systemctl stop nginx
+   ```
+
+## Adding Coder deployment subdomain
+
+> This example assumes Coder is running locally on `127.0.0.1:3000` for the subdomain `YOUR_SUBDOMAIN` e.g. `coder.example.com`.
+
+1. Create NGINX configuration for this app:
+
+   ```console
+   sudo touch /etc/nginx/sites-available/YOUR_SUBDOMAIN
+   ```
+
+2. Activate this file:
+
+   ```console
+   sudo ln -s /etc/nginx/sites-available/YOUR_SUBDOMAIN /etc/nginx/sites-enabled/YOUR_SUBDOMAIN
+   ```
+
+## Install and configure LetsEncrypt Certbot
+
+1. Install LetsEncrypt Certbot: Refer to the [CertBot documentation](https://certbot.eff.org/instructions?ws=other&os=pip&tab=wildcard)
+
+## Create DNS provider credentials
+
+1. Create an API token for the DNS provider you're using: e.g [CloudFlare](https://dash.cloudflare.com/profile/api-tokens) with the following permissions:
+
+   - Zone - DNS - Edit
+
+2. Create a file in `.secrets/certbot/cloudflare.ini` with the following content:
+
+   ```ini
+   dns_cloudflare_api_token = YOUR_API_TOKEN
+   ```
```

Comment:
Can you make the top-level step something more generic? I'm concerned people with another DNS provider will get lost at this step.

> Follow the [Certbot DNS Plugins](https://eff-certbot.readthedocs.io/en/stable/using.html#dns-plugins) documentation to configure Certbot for your DNS provider. This is required to generate a wildcard certificate.

Then provide the example for Cloudflare (maybe just under one step)?

Also, what do you think about providing a command for the file creation? I noticed in other cases you used `touch`. For example:

```console
mkdir -p ~/.secrets/certbot
touch ~/.secrets/certbot/cloudflare.ini
vim ~/.secrets/certbot/cloudflare.ini
```

Reply:
I agree. I will update.